File: app/Modules/Shifts/Models/LeaveRequest.php
<?php

namespace App\Modules\Shifts\Models;

use App\Models\User;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class LeaveRequest extends Model
{
    use HasFactory, SoftDeletes;

    protected $table = 'hr_leave_requests';

    protected $fillable = [
        'employee_id', 'type', 'start_date', 'end_date',
        'start_half_day', 'end_half_day', 'reason', 'status',
        'validated_by', 'validated_at', 'rejection_reason',
    ];

    protected function casts(): array
    {
        return [
            'start_date'   => 'date',
            'end_date'     => 'date',
            'validated_at' => 'datetime',
        ];
    }

    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class);
    }

    public function validator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'validated_by');
    }

    public function scopePending($query)
    {
        return $query->where('status', 'pending');
    }

    public function scopeApproved($query)
    {
        return $query->where('status', 'approved');
    }
}

File: mobile/app/src/main/java/com/chronosphere/mobile/models/LeaveRequest.java
package com.chronosphere.mobile.models;

import com.google.gson.annotations.SerializedName;

public class LeaveRequest {
    public int    id;

    @SerializedName("employee_id")
    public int employeeId;

    @SerializedName("type")
    public String leaveType;

    @SerializedName("start_date")
    public String startDate;

    @SerializedName("end_date")
    public String endDate;

    public String reason;
    public String status;

    @SerializedName("rejection_reason")
    public String rejectionReason;

    public EmployeeInfo employee;

    public static class EmployeeInfo {
        public int    id;
        public String name;
        public String code;
    }
}
